### DISCUSSION TITLE
Nested and visual hierarchy syntax like Sass

### DISCUSSION THREAD

**Discussion by 1aron** [master-co/css#148] [Ideas]
With Master CSS as a language, I'm thinking about the possibility of a Sass-like syntax structure to:

- Deduplicate property names
- Deduplicate selectors
- Deduplicate media queries
- The `.scss` structure you're familiar with

<img width="1219" alt="Xnapper-2023-01-17-16 25 42" src="https://user-images.githubusercontent.com/33840671/212846578-13ddecb6-b61c-484a-b5ee-a8d33323d82f.png">

**Comment by abdo-spices** [comment]
if you provided a [macaron](https://github.com/macaron-css/macaron) like api for building UI libraries and if you did 
add tools to makes making universal components easier

**Reply by 1aron** [reply]
We have also researched a similar solution https://github.com/joe-bell/cva. Our goal is to quickly create styling elements in a more concise way through the syntax of Master CSS.
- https://github.com/master-co/style-element/issues/6#issuecomment-1304209339
